### PR TITLE
doc: add code of conduct prefix text

### PR DESCRIPTION
**Pull request by anonrig**
For future tracking here's the steps taken/will be taken for the patching script:

This pull request includes the initial changes required to have a patching script for the CoC template. The patching script basically get's a commit SHA, and appends the original CoC template (available from code covenants website), and later applies the patching file (which will be included in the next pull request adding the script).

Steps needed to take to add the patching script:

1. Add a PREAMBLE file containing only the first portion of CoC
  - This pull request copies the first portion, and later applies changes only required to make sense in this context. (Removing some links and adding context to CoC modifications) 
2. Add a patching script (a different pull request)
  - Update the `base_url` on the patch script with the URL of the SHA commit from this pull request to use it as a baseline. 
  - Remove the CODE_OF_CONDUCT.md
  - Generate an initial patching file required to have the current version of CODE_OF_CONDUCT.md with minimal changes.
  - Push it to the patching script.
  
Co-authored by @joesepi

### CODE REVIEW

**Review by ljharb — APPROVED** [review]

**Comment by bnb** [comment]
can you describe what you mean by "prefix"? 🤔

**Review by mhdawson — APPROVED** [review]
LGTM

**Comment by ljharb** [comment]
I read it to mean that the CoC content will be appended after the prefix text.

**Comment by bnb** [comment]
if that is the case, we probably want to change the wording in the first section to reflect that we use a modified version and that the translations are available but don't reflect the modified version, right?

**Review by dylans — APPROVED** [review]

File: conduct/artifacts/CODE_OF_CONDUCT_PREFIX.md
Comment (bnb) [inline conduct/artifacts/CODE_OF_CONDUCT_PREFIX.md:3]: this should reflect that we use a modified version of the Contributor Covenant and that the translations do not reflect those modifications.

**Review by timmywil — APPROVED** [review]

File: conduct/artifacts/CODE_OF_CONDUCT_PREFIX.md
Comment (bnb) [inline conduct/artifacts/CODE_OF_CONDUCT_PREFIX.md:10]: this should either be a paragraph or bullets, not both together. Fine with either suggestion or alternatives: 

```suggestion
All recipients of reports commit to:

- maintain the confidentiality with regard to the reporter and victim of an incident
- participate in the path for escalation as outlined in the section on Escalation when required
```


```suggestion
All recipients of reports commit to maintain the confidentiality with regard to the reporter and victim of an incident, and participate in the path for escalation as outlined in the section on Escalation when required.
```

File: conduct/artifacts/CODE_OF_CONDUCT_PREFIX.md
Comment (bnb) [inline conduct/artifacts/CODE_OF_CONDUCT_PREFIX.md:35]: Potentially want to clarify here that this isn't meant to duplicate the above section on reporting an issue in a project. On first read, that wasn't clear to me and I've been involved in this stuff for a long time - I imagine it might be even less clear to people who aren't familiar with this.

**Review by bnb — CHANGES_REQUESTED** [review]

File: conduct/artifacts/CODE_OF_CONDUCT_PREFIX.md
Comment (tobie) [reply conduct/artifacts/CODE_OF_CONDUCT_PREFIX.md:35]: Wait, the CoCP should handle both reports for projects who don't have their own process AND escalations.

**Review by tobie — CHANGES_REQUESTED** [review]
I'm confused about whether this PR is changing the content of the CoC or just moving it around and trimming the imported part.

If it's the latter I suggest having to separate commits, one that moves the CoC, and the other one that amends it, so it it clear what's going on.

**Comment by tobie** [comment]
@bnb [wrote](#issuecomment-1371284587):
> can you describe what you mean by "prefix"? 🤔

I think we should rename this file to CODE_OF_CONDUCT_PREAMBLE.md or CODE_OF_CONDUCT_INTRO.md. That would be clearer.

File: conduct/artifacts/CODE_OF_CONDUCT_PREFIX.md
Comment (Relequestual) [reply conduct/artifacts/CODE_OF_CONDUCT_PREFIX.md:10]: Bulets make it easier to add to should such adding be required.

**Comment by anonrig** [comment]
I created two commits, the first moving the file and the second making the changes. After the meeting, I'll update the changes and re-request for reviews. Thank you for all the patience in the review of this pull request.

**Comment by anonrig** [comment]
> I think we should rename this file to CODE_OF_CONDUCT_PREAMBLE.md or CODE_OF_CONDUCT_INTRO.md. That would be clearer.

@tobie I've changed the naming of the file. Thank you for the suggestion!

File: conduct/artifacts/CODE_OF_CONDUCT_PREFIX.md
Comment (anonrig) [reply conduct/artifacts/CODE_OF_CONDUCT_PREFIX.md]: I've applied the recommended changes. Thank you for the feedback.

File: conduct/artifacts/CODE_OF_CONDUCT_PREFIX.md
Comment (anonrig) [reply conduct/artifacts/CODE_OF_CONDUCT_PREFIX.md]: Should we make these changes after the initial document is passed, and merged? I think it will reduce friction and will make the next steps easier. Depending on preference, I can make the suggestions here as an additional commit.

**Review by tobie — APPROVED** [review]
Thanks for those changes. This is super helpful. As discussed in the call, it would be great if we had a summary in the body of the commit of why we're adding the COC_PREAMBLE rather than moving the COC.

**Comment by anonrig** [comment]
Thanks for the review. I've updated/added a description for this pull request. I hope it clarifies a couple of things. @tobie

**Review by ljharb — APPROVED** [review]

File: conduct/artifacts/CODE_OF_CONDUCT_PREFIX.md
Comment (anonrig) [reply conduct/artifacts/CODE_OF_CONDUCT_PREFIX.md]: I just added a commit to address this issue. Appreciate if you can review it, thank you.

File: conduct/artifacts/CODE_OF_CONDUCT_PREFIX.md
Comment (anonrig) [reply conduct/artifacts/CODE_OF_CONDUCT_PREFIX.md]: The recommendation was ambiguous (for a non-native speaker like me). If you're concerned and have a specific recommendation, can you add this to this review, or can we do it in a different pull request?

Comment (bnb) [reply conduct/artifacts/CODE_OF_CONDUCT_PREFIX.md]: this is unresolved but shouldn't block merging.

**Review by bnb — APPROVED** [review]

**Pull request merged**
